#5 予定作成ページの実装

MaekawaShun331 opened the pull request:
・予定作成機能を実装しました
・uuidのライブラリを組み込みました
・SQL発行時にログを出力するよう修正しました
・Scheduleモデルで2箇所設定ミスがあったので修正しました

MaekawaShun331 committed:
#5 予定作成ページの実装
・予定作成機能を実装しました
・uuidのライブラリを組み込みました
・SQL発行時にログを出力するよう修正しました
・Scheduleモデルで2箇所設定ミスがあったので修正しました

MaekawaShun331 committed:
#5 予定作成ページの実装
・バリデーション実装

jtakakura commented on app/Http/Controllers/ScheduleController.php:
```diff
@@ -0,0 +1,137 @@
+        $now = Carbon::now();
+        //候補日をスケジュールIDと合わせてレコード化する
+        $candidates = [];
+        foreach ($candidateNames as $canName) {
```
@MaekawaShun331 ここは `array_map` で書けそうな気がするので試して見てください。

MaekawaShun331 replied:
@jtakakura  array_mapで書けそうと考えていたのですが、array_mapで実行する関数にどうやって$scheduleを渡せば良いか分からなかったです。  
渡す方法あるでしょうか？

jtakakura replied:
@MaekawaShun331 `use` というキーワードで調べてみてください。

jtakakura commented on app/Http/Controllers/ScheduleController.php:
```diff
@@ -0,0 +1,137 @@
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
```
@MaekawaShun331 今回の実装としては問題ないのですが、 Laravel には `FormRequest` という機能がありますので、今後更新系とか実装するにあたり、現状のバリデーションの実装がいいか、上記 `FormRequest` を使うのがいいか、検討して見てください。
(機能としては調査して、ざっくりとでも理解してもらえればと思います)

MaekawaShun331 replied:
@jtakakura 確認しておきます。

jtakakura commented on app/Http/Controllers/ScheduleController.php:
```diff
@@ -0,0 +1,137 @@
+        }
+
+        //入力された内容で候補日を登録する
+        DB::table('candidates')->insert($candidates);
```
@MaekawaShun331 1点気になったのですが、 `candidate` への INSERT で失敗した場合、先に登録したスケジュール情報はどうなりますか?

MaekawaShun331 replied:
@jtakakura 考慮していませんでした。  
エラー起こったらスケジュール情報だけ残る事になると思います。
対策しておきます。

MaekawaShun331 committed:
#5 レビュー指摘内容反映
・例外処理を実装しました
・入力された候補日をレコード化する際に、array_mapを使ってレコード化するよう修正しました
・設定ファイルの内容を変更しました

jtakakura commented on app/Http/Controllers/ScheduleController.php:
```diff
@@ -56,38 +56,46 @@ public function store(Request $request)
-                'created_at' => $now,
-                'updated_at' => $now
-            ];
+        DB::beginTransaction();
```
@MaekawaShun331 `transaction` メソッドを使って実装してもらっていいですか?

https://qiita.com/sola-msr/items/29c2bb039116dfee0834#db%E3%83%95%E3%82%A1%E3%82%B5%E3%83%BC%E3%83%89%E3%81%AEtransaction%E3%83%A1%E3%82%BD%E3%83%83%E3%83%89%E3%82%92%E4%BD%BF%E7%94%A8%E3%81%99%E3%82%8B%E6%96%B9%E6%B3%95

MaekawaShun331 committed:
#5 レビュー指摘内容反映2 トランザクション処理で`transaction` メソッドを使うよう修正

jtakakura reviewed (approved):
LGTM
